### PR TITLE
populate eslint with common rules

### PR DESCRIPTION
**Pull request by alexlamsl**
From https://github.com/kangax/html-minifier/commit/560e109e7f3fba6882b08c9d760404fc28754362#commitcomment-16783499

@XhmikosR I've incorporated the rules from https://github.com/giakki/uncss/blob/deps/.eslintrc - the omitted ones are almost all due to being included in the `recommended` set.

### CODE REVIEW

**Comment by alexlamsl** [comment]
Almost forgot - I've left `strict` out since I couldn't get the declaration style configured. Let's add that rule after #569

File: dist/htmlminifier.js [hunk 1]
```diff
@@ -1698,7 +1698,7 @@
     return minify(value, options);
   };
 
-}(typeof exports === 'undefined' ? this : exports));
+})(typeof exports === 'undefined' ? this : exports);
```
Comment (XhmikosR) [inline dist/htmlminifier.js:1701]: We can go with `outside` instead.

File: .eslintrc.json [hunk 2]
```diff
@@ -11,7 +12,88 @@
...
+    "comma-spacing": "error",
+    "comma-style": [
+      "error",
+      "last"
+    ],
+    "curly": "error",
+    "eol-last": "error",
+    "eqeqeq": [
+      "error",
+      "smart"
```
Comment (XhmikosR) [inline .eslintrc.json]: I wouldn't go with smart.

**Comment by XhmikosR** [comment]
Are you sure none of the added rules are not already covered by the eslint recommended?

File: tests/minifier.js [hunk 3]
```diff
@@ -777,21 +777,21 @@
...
-             '<input type=\"text\">\r\n<textarea><\/textarea>\r\n<pre><\/pre>';
-    output = '<script type="text/javascript">var = "hello";</script>'                                +
-             '<style type="text/css">#foo { color: red;        }</style>'                            +
-             '<div><div><div>'                                                                       +
-             '<!-- hello --><div><!--! hello --><div><div class="">'                                 +
-             '<textarea disabled="disabled">     this is a textarea </textarea>'                     +
-             '</div></div></div></div></div></div>'                                                  +
-             '<pre>       \r\nxxxx</pre><span>x</span> <span>Hello</span> <b>billy</b> '             +
+    input = '<script type=\"text\/javascript\">var = \"hello\";<\/script>\r\n\r\n\r\n' +
+            '<style type=\"text\/css\">#foo { color: red;        }          <\/style>\r\n\r\n\r\n' +
```
Comment (XhmikosR) [inline tests/minifier.js:781]: Maybe remove or align the lines before +.

**Comment by alexlamsl** [comment]
Ah, I should mention this - I look up http://eslint.org/docs/rules/ as I scan through your list. So it's a matter of skipping the ones with a big tick next to it.

File: tests/minifier.js (hunk 3 above)
Comment (alexlamsl) [inline tests/minifier.js:781]: Sorry I don't quite follow what you mean...?

File: .eslintrc.json (hunk 2 above)
Comment (alexlamsl) [inline .eslintrc.json]: There's `== null` in `cli.js` and I'm not sure if I can get rid of that safely...

File: dist/htmlminifier.js (hunk 1 above)
Comment (alexlamsl) [inline dist/htmlminifier.js:1701]: I'll have to edit some files either way `htmlparser.js` is `inline` - but `outside` it is then :wink:

File: .eslintrc.json (hunk 2 above)
Comment (XhmikosR) [inline .eslintrc.json]: Just add an ignore for that file then.

File: dist/htmlminifier.js (hunk 1 above)
Comment (XhmikosR) [inline dist/htmlminifier.js:1701]: I don't mind which one, I just thought we were using outside already. So, just keep it as you have it here.

File: tests/minifier.js (hunk 3 above)
Comment (XhmikosR) [inline tests/minifier.js:781]: It was aligned before, no big deal.

Comment (alexlamsl) [inline tests/minifier.js:781]: I have to change these due to `no-multi-spaces` complaining - and it's the only place in the project that rule has a trouble with...

**Comment by alexlamsl** [comment]
@XhmikosR addressed your comments except those `+` lines inside `tests/minifier.js` - anything else?

**Comment by XhmikosR** [comment]
Looks good to me. I'd just squash the patches and merge.

**Comment by alexlamsl** [comment]
Will do~ :tada:

**Comment by XhmikosR** [comment]
Maybe keep `inside`? It was used in more places I believe.

We should also ignore the Google Analytics snippet.

**Comment by alexlamsl** [comment]
Oh alright then - Google Analytics is already `inside`, so that's even better :sunglasses:

**Comment by alexlamsl** [comment]
Great timing - my `git push -f` failed due to Github throwing Angry Unicorns at me...

**Pull request merged**